OpenXR plugin requires too new C++ standard library

Ubuntu 22.04, which is our new target starting this month, comes with GCC 11.4. However, the OpenXR plugin seems to make use of std::format, which [apparently requires GCC 13](https://stackoverflow.com/questions/63724059/does-gcc-support-c20-stdformat). I worked around this issue by upgrading to GCC 13 on Ubuntu 22.04, but it turns out that this doesn't work; While it builds fine, it makes use of new GLIBCXX version features that are not available on a normal Ubuntu 22.04 installation, meaning that users need a newer operating system. Currently, our policy is to support the oldest Ubuntu release that still receives standard support from Canonical (same as AppImageHub's policy).

I wouldn't be surprised if std::format usage in the OpenXR plugin is the only thing which keeps us from staying on GCC 11.4, since GCC 11.4 should support most other C++20 features.
```gcc
/root/overte/plugins/openxr/src/OpenXrDisplayPlugin.cpp:19:10: fatal error: format: No such file or directory
   19 | #include <format>
      |          ^~~~~~~~
```

--- a/plugins/openxr/src/OpenXrDisplayPlugin.cpp
+++ b/plugins/openxr/src/OpenXrDisplayPlugin.cpp
@@ -16,7 +16,7 @@
 #include <glm/gtx/string_cast.hpp>
 #include <glm/gtx/transform.hpp>
 #include <thread>
-#include <format>
+#include <sstream>
 
 #if defined(Q_OS_WIN)
 #undef near
@@ -133,8 +133,11 @@ static std::string glFormatStr(GLenum source) {
         ENUM_TO_STR(GL_RGBA16F);
         ENUM_TO_STR(GL_SRGB8_ALPHA8);
         ENUM_TO_STR(GL_RGB10_A2UI);
-        default:
-            return std::format("0x{:X}", source);
+        default: {
+            std::stringstream stream;
+            stream << "0x" << std::uppercase << std::hex << source;
+            return stream.str();
+        }
     }
 }
 

